## Hypothesis Testing - Assessment

In [1]:
cd"F:\Data Science\AnalytixLab\Classes\DSP Class 12-13 files\DSP Class 12-13 files"

F:\Data Science\AnalytixLab\Classes\DSP Class 12-13 files\DSP Class 12-13 files


In [2]:
pwd

'F:\\Data Science\\AnalytixLab\\Classes\\DSP Class 12-13 files\\DSP Class 12-13 files'

In [3]:
import pandas as pd

In [7]:
import scipy.stats as stats

In [5]:
dataset=pd.read_csv("dietstudy.csv")

In [6]:
dataset

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0       1   45    Male  180  148  106  113  100   198   196   193   188   192
1       2   56    Male  139   94  119   75   92   237   233   232   228   225
2       3   50    Male  152  185   86  149  118   233   231   229   228   226
3       4   46  Female  112  145  136  149   82   179   181   177   174   172
4       5   64    Male  156  104  157   79   97   219   217   215   213   214
5       6   49  Female  167  138   88  107  171   169   166   165   162   161
6       7   63    Male  138  132  146  143  132   222   219   215   215   210
7       8   63  Female  160  128  150  118  123   167   167   166   162   161
8       9   52    Male  107  120  129  195  174   199   200   196   196   193
9      10   45    Male  156  103  126  135   92   233   229   229   229   226
10     11   61  Female   94  144  114  114  121   179   181   176   173   173
11     12   49  Female  107   93  156  148  150   158   153   155   155   154
12     13   61  Female  145  107  129   86  159   157   151   150   145   143
13     14   59    Male  186  142  128  122  101   216   213   210   210   206
14     15   52    Male  112  107  103   89  148   257   255   254   252   249
15     16   60  Female  104  103  117   79  130   151   146   144   144   140

#### 1. A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed (Data set: dietstudy.csv) 

###### Hypothesis testing for weight:

H0: weight before = weight after

H1: weight before > weight after

So since there is a before after scenario, we would conduct a paired sample t-test

In [8]:
stats.ttest_rel(a=dataset.wgt0,b=dataset.wgt4)

Ttest_relResult(statistic=11.174521688532522, pvalue=1.137689414996614e-08)

Since we are conducting one tailed test and pvalue/2 <0.05.Thus we reject the null and since sign of statistics is (+), a>b .So we conclude that weight before was greater than weight after with probability=1-(pvalue/2) under confidence level of 95% and type 1 error(alpha)=0.05

###### Hypothesis testing for triglycreide level:

H0: triglycreide before = triglycreide after

H1: triglycreide before > triglycreide after

So since there is a before after scenario, we would conduct a paired sample t-test

In [9]:
stats.ttest_rel(a=dataset.tg0,b=dataset.tg4)

Ttest_relResult(statistic=1.2000008533342437, pvalue=0.24874946576903698)

Since pvalue >0.05, we do not reject the null hypothesis and conclude that there is no difference between the before and after triglyceride level

#### 2.An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales? (Data set: creditpromo.csv) 

In [11]:
dataset=pd.read_csv('creditpromo.csv')

In [14]:
dataset.sample(10)

id         insert      dollars  value
333   90396  New Promotion  1873.206608      2
233   60156  New Promotion  1424.089762      2
394  106180  New Promotion  1520.601738      2
65    19586  New Promotion  1283.318958      2
302   80037  New Promotion  1808.377839      2
2       973       Standard  2327.092181      1
19     5126       Standard  1567.241966      1
68    21162  New Promotion  1668.658645      2
213   56231  New Promotion  1417.741896      2
41    12494       Standard  1987.125609      1

In [23]:
dataset.value.unique()

array([1, 2], dtype=int64)

In [16]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 4 columns):
id         500 non-null int64
insert     500 non-null object
dollars    500 non-null float64
value      500 non-null int64
dtypes: float64(1), int64(2), object(1)
memory usage: 15.7+ KB


###### Hypothesis testing for promotion effectiveness:

H0: dollar spend with old promotion = dollar spend with new promotion

H1: dollar spend with old promotion < dollar spend with new promotion

So since there are two different samples that need to be compared(one with old promotion and other with new promotion), we would conduct a two sample t-test (independent)

In [24]:
stats.ttest_ind(a=dataset.dollars[dataset.value==2],
               b=dataset.dollars[dataset.value==1],
               equal_var=False)

Ttest_indResult(statistic=2.2604227260537053, pvalue=0.024226348217341587)

From the dataset dataset.value==2 indicates "New Promotion" and dataset.value==1 indicates "Standard promotion"

Since we are conducting a one tailed test , we can see that pvalue/2 <0.05 and sign of statistics is +ve a>b i.e dollars spend after new promotion > dollars spend in standard promotion

Hence we conclude that promotion was effective with a probability=1-(pvalue/2) under confidence level of 95% and type 1 error(alpha)=0.05

#### 3.An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). (Data set: pollination.csv) 

a. Is the overall population of Seed yield/plant (g) equals to 200? 

b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different. 

In [25]:
dataset=pd.read_csv('pollination.csv')

In [28]:
dataset

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0   Natural      1.85            147.70            16.86
1   Natural      1.86            136.86            16.77
2   Natural      1.83            149.97            16.35
3   Natural      1.89            172.33            18.26
4   Natural      1.80            144.46            17.90
5   Natural      1.88            138.30            16.95
6   Natural      1.89            150.58            18.15
7   Natural      1.79            140.99            18.86
8   Natural      1.85            140.57            18.39
9   Natural      1.84            138.33            18.58
10     Hand      2.58            224.26            18.18
11     Hand      2.74            197.50            18.07
12     Hand      2.58            230.34            19.07
13     Hand      2.62            217.05            19.00
14     Hand      2.68            233.84            18.00
15     Hand      2.56            216.52            18.49
16     Hand      2.34            211.93            17.45
17     Hand      2.67            210.37            18.97
18     Hand      2.45            199.87            19.31
19     Hand      2.44            214.30            19.36

a. Is the overall population of Seed yield/plant (g) equals to 200? 

###### Hypothesis testing for Seed yield/plant (g) equals to 200:

H0: Seed yield/plant (g) = 200

H1: Seed yield/plant (g) != 200

Since we have population mean available (200) we will conduct one sample t-test


In [29]:
stats.ttest_1samp(a=dataset.Seed_Yield_Plant,
                 popmean= 200)

Ttest_1sampResult(statistic=-2.3009121248548645, pvalue=0.032891040921283025)

Since it is a two tailed test and pvalue is <0.05, we reject the null

Hence we can conclude that Seed yield/plant (g) != 200 with a probability=1-pvalue under the confidence level of 95% and type 1 error(alpha)=0.05

b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different. 

###### Hypothesis testing  for effectiveness under natural pollination and hand pollination:

##### case 1

H0: natural pollination fruit wt = hand pollination fruit wt

H1: natural pollination fruit wt != hand pollination fruit wt

two sample t test (independent)


##### case 2

H0: natural pollination Seed yield/plant (g) = hand pollination Seed yield/plant (g)

H1: natural pollination Seed yield/plant (g)= hand pollination Seed yield/plant (g)

two sample t test (independent)


##### case 3

H0: natural pollination Seedling_length = hand pollination Seedling_length

H1: natural pollination Seedling_length= hand pollination Seedling_length

two sample t test (independent)

In [32]:
##case 1:
stats.ttest_ind(a=dataset.Fruit_Wt[dataset.Group=="Natural"],
               b=dataset.Fruit_Wt[dataset.Group=="Hand"],
               equal_var=False)

Ttest_indResult(statistic=-17.669989614440286, pvalue=4.306871213074868e-09)

Since its a two tailed test and pvaule is <0.05, hence we reject the null

Thus we can conclude that Natural and Hand Pollination wrt fruit wt are different with a probability=1-pvalue under confidence level 95% and type 1 error(alpha)=0.05

The sign of statistics is -ve and pvalue/2 is also <0.05, thus b>a i.e hand pollination is more effective than natural pollination wrt to fruit wt

In [33]:
##case 2:
stats.ttest_ind(a=dataset.Seed_Yield_Plant[dataset.Group=="Natural"],
               b=dataset.Seed_Yield_Plant[dataset.Group=="Hand"],
               equal_var=False)

Ttest_indResult(statistic=-13.958260515902547, pvalue=5.136161282685624e-11)

Since its a two tailed test and pvaule is <0.05, hence we reject the null

Thus we can conclude that Natural and Hand Pollination wrt Seed_Yield_Plant are different with a probability=1-pvalue under confidence level 95% and type 1 error(alpha)=0.05

The sign of statistics is -ve and pvalue/2 is also <0.05, thus b>a i.e hand pollination is more effective than natural pollination wrt to Seed_Yield_Plant

In [35]:
##case 3:
stats.ttest_ind(a=dataset.Seedling_length[dataset.Group=="Natural"],
               b=dataset.Seedling_length[dataset.Group=="Hand"],
               equal_var=False)

Ttest_indResult(statistic=-2.542229999657055, pvalue=0.021430608378161634)

Since its a two tailed test and pvaule is <0.05, hence we reject the null

Thus we can conclude that Natural and Hand Pollination wrt Seedling_length are different with a probability=1-pvalue under confidence level 95% and type 1 error(alpha)=0.05

The sign of statistics is -ve and pvalue/2 is also <0.05, thus b>a i.e hand pollination is more effective than natural pollination wrt to Seedling_length

#### 4.An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently? (Data set: dvdplayer.csv). 

In [36]:
dataset=pd.read_csv("dvdplayer.csv")

In [38]:
dataset.head(10)

Age_group   dvdscore
0  65_and_over  38.454803
1     55_to_64  17.669677
2  65_and_over  31.704307
3  65_and_over  25.924460
4     Under_25  30.450007
5     Under_25  35.609909
6  65_and_over  29.677695
7     25_to_34  38.167369
8  65_and_over  23.509700
9     45_to_54  26.051029

###### Hypothesis testing  design rating by various age group:

H0: Rating by all age group are same

H1: Rating by atleast one age group is different

Hence we can use ANOVA here

In [39]:
dataset.Age_group.unique()

array(['65_and_over', '55_to_64', 'Under_25', '25_to_34', '45_to_54',
       '35_to_44'], dtype=object)

In [41]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 68 entries, 0 to 67
Data columns (total 2 columns):
Age_group    68 non-null object
dvdscore     68 non-null float64
dtypes: float64(1), object(1)
memory usage: 1.1+ KB


In [43]:
a=dataset.dvdscore[dataset.Age_group=='65_and_over']
b=dataset.dvdscore[dataset.Age_group=='55_to_64']
c=dataset.dvdscore[dataset.Age_group=='Under_25']
d=dataset.dvdscore[dataset.Age_group=='25_to_34']
e=dataset.dvdscore[dataset.Age_group=='45_to_54']

In [44]:
stats.f_oneway(a,b,c,d,e)

F_onewayResult(statistic=7.361779935535184, pvalue=9.263912804562245e-05)

Since its a two tailed test and pvalue is <0.05,we reject the null

Hence we conclude that atleast one of the age group rated differently with probability=1-pvalue under confidence level 95% and type 1 error(alpha)=0.05

#### 5.A survey was conducted among 2800 customers on several demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose. (Data set: sample_survey.csv).  
a. Is there any relationship in between labour force status with marital status? 

b. Do you think educational qualification is somehow controlling the marital status? 

c. Is happiness is driven by earnings or marital status? 

In [72]:
dataset=pd.read_csv("sample_survey.csv")

In [73]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2832 entries, 0 to 2831
Data columns (total 46 columns):
id          2832 non-null int64
wrkstat     2831 non-null object
marital     2831 non-null object
childs      2825 non-null float64
age         2828 non-null float64
educ        2820 non-null float64
paeduc      2041 non-null float64
maeduc      2399 non-null float64
speduc      1311 non-null float64
degree      2822 non-null object
sex         2832 non-null object
race        2832 non-null object
born        2819 non-null object
parborn     2812 non-null object
granborn    2630 non-null float64
income      2503 non-null object
rincome     1849 non-null object
polviews    2691 non-null object
cappun      2599 non-null object
postlife    2066 non-null object
happy       2806 non-null object
hapmar      1337 non-null object
owngun      1869 non-null object
news        1870 non-null object
tvhours     2337 non-null float64
howpaid     1347 non-null object
ethnic      2238 non-null ob

In [74]:
dataset.isnull().sum()

id             0
wrkstat        1
marital        1
childs         7
age            4
educ          12
paeduc       791
maeduc       433
speduc      1521
degree        10
sex            0
race           0
born          13
parborn       20
granborn     202
income       329
rincome      983
polviews     141
cappun       233
postlife     766
happy         26
hapmar      1495
owngun       963
news         962
tvhours      495
howpaid     1485
ethnic       594
eth1         275
eth2        1730
eth3        2388
confinan     968
conbus      1011
coneduc      951
conpress     970
conmedic     957
contv        956
agecat         4
childcat       7
news1          0
news2          0
news3          0
news4          0
news5          0
car1           0
car2         566
car3        1160
dtype: int64

In [84]:
dataset.head(20)

id               wrkstat        marital  childs   age  educ  paeduc  \
0    1     Working full time       Divorced     2.0  60.0  12.0    12.0   
1    2     Working part-time  Never married     0.0  27.0  17.0    20.0   
2    3     Working full time        Married     2.0  36.0  12.0    12.0   
3    4     Working full time  Never married     0.0  21.0  13.0     NaN   
4    5     Working full time  Never married     0.0  35.0  16.0     NaN   
5    6     Working full time       Divorced     1.0  33.0  16.0     9.0   
6    7     Working full time      Separated     0.0  43.0  12.0    14.0   
7    8     Working full time  Never married     0.0  29.0  13.0    16.0   
8    9     Working part-time        Married     2.0  39.0  18.0    16.0   
9   10     Working full time       Divorced     0.0  45.0  15.0    16.0   
10  11  Unemployed, laid off  Never married     0.0  29.0  12.0    12.0   
11  12     Working full time        Married     1.0  41.0  15.0     NaN   
12  13     Working part-time       Divorced     2.0  32.0  14.0     NaN   
13  14     Working full time        Married     1.0  48.0  20.0    12.0   
14  15         Keeping house  Never married     0.0  20.0  12.0    12.0   
15  16     Working full time        Married     5.0  43.0  16.0     6.0   
16  17     Working full time       Divorced     4.0  27.0  11.0     6.0   
17  18         Keeping house        Widowed     7.0  34.0   7.0     NaN   
18  19     Working full time      Separated     0.0  43.0   9.0     0.0   
19  20     Working full time        Married     1.0  28.0  16.0    16.0   

    maeduc  speduc          degree    ...           agecat   childcat news1  \
0     12.0     NaN     High school    ...         55 to 64     01-Feb    No   
1      NaN     NaN  Junior college    ...         25 to 34       None    No   
2     12.0    16.0     High school    ...         35 to 44     01-Feb    No   
3     12.0     NaN     High school    ...     Less than 25       None    No   
4     12.0     NaN        Bachelor    ...         35 to 44       None    No   
5      6.0     NaN        Bachelor    ...         25 to 34     01-Feb    No   
6     12.0     NaN     High school    ...         35 to 44       None    No   
7     12.0     NaN     High school    ...         25 to 34       None    No   
8     12.0    13.0        Bachelor    ...         35 to 44     01-Feb   Yes   
9     12.0     NaN  Junior college    ...         45 to 54       None    No   
10    12.0     NaN     High school    ...         25 to 34       None    No   
11     8.0    14.0     High school    ...         35 to 44     01-Feb    No   
12    12.0     NaN     High school    ...         25 to 34     01-Feb    No   
13     8.0    16.0        Bachelor    ...         45 to 54     01-Feb    No   
14    12.0     NaN     High school    ...     Less than 25       None    No   
15     6.0    12.0        Bachelor    ...         35 to 44  5 or more    No   
16     4.0     NaN  LT High school    ...         25 to 34     03-Apr    No   
17     NaN     NaN  LT High school    ...         25 to 34  5 or more    No   
18     NaN     NaN  LT High school    ...         35 to 44       None    No   
19    16.0    16.0        Bachelor    ...         25 to 34     01-Feb    No   

   news2  news3 news4 news5      car1      car2      car3  
0     No     No    No    No  American  Japanese  Japanese  
1     No    Yes    No    No  American    German  Japanese  
2     No     No   Yes   Yes  American  American       NaN  
3     No     No   Yes   Yes  American     Other       NaN  
4     No     No    No    No  American  American    Korean  
5    Yes     No   Yes   Yes  American    Korean  Japanese  
6     No     No   Yes   Yes    Korean  American  American  
7     No     No   Yes   Yes     Other  Japanese       NaN  
8     No    Yes    No    No  American    Korean       NaN  
9    Yes     No   Yes   Yes    Korean  Japanese    Korean  
10   Yes    Yes    No   Yes    Korean  Japanese  Japanese  
11    No     No    No    No  Japanese    German 

In [83]:
dataset.marital.mode()[0]

'Married'

In [80]:
dataset.marital.fillna(dataset.marital.mode()[0],inplace=True)

In [85]:
dataset.wrkstat.fillna(dataset.wrkstat.mode()[0],inplace=True)

In [89]:
dataset.columns

Index(['id', 'wrkstat', 'marital', 'childs', 'age', 'educ', 'paeduc', 'maeduc',
       'speduc', 'degree', 'sex', 'race', 'born', 'parborn', 'granborn',
       'income', 'rincome', 'polviews', 'cappun', 'postlife', 'happy',
       'hapmar', 'owngun', 'news', 'tvhours', 'howpaid', 'ethnic', 'eth1',
       'eth2', 'eth3', 'confinan', 'conbus', 'coneduc', 'conpress', 'conmedic',
       'contv', 'agecat', 'childcat', 'news1', 'news2', 'news3', 'news4',
       'news5', 'car1', 'car2', 'car3'],
      dtype='object')

In [91]:
dataset.head(20)

id               wrkstat        marital  childs   age  educ  paeduc  \
0    1     Working full time       Divorced     2.0  60.0  12.0    12.0   
1    2     Working part-time  Never married     0.0  27.0  17.0    20.0   
2    3     Working full time        Married     2.0  36.0  12.0    12.0   
3    4     Working full time  Never married     0.0  21.0  13.0     NaN   
4    5     Working full time  Never married     0.0  35.0  16.0     NaN   
5    6     Working full time       Divorced     1.0  33.0  16.0     9.0   
6    7     Working full time      Separated     0.0  43.0  12.0    14.0   
7    8     Working full time  Never married     0.0  29.0  13.0    16.0   
8    9     Working part-time        Married     2.0  39.0  18.0    16.0   
9   10     Working full time       Divorced     0.0  45.0  15.0    16.0   
10  11  Unemployed, laid off  Never married     0.0  29.0  12.0    12.0   
11  12     Working full time        Married     1.0  41.0  15.0     NaN   
12  13     Working part-time       Divorced     2.0  32.0  14.0     NaN   
13  14     Working full time        Married     1.0  48.0  20.0    12.0   
14  15         Keeping house  Never married     0.0  20.0  12.0    12.0   
15  16     Working full time        Married     5.0  43.0  16.0     6.0   
16  17     Working full time       Divorced     4.0  27.0  11.0     6.0   
17  18         Keeping house        Widowed     7.0  34.0   7.0     NaN   
18  19     Working full time      Separated     0.0  43.0   9.0     0.0   
19  20     Working full time        Married     1.0  28.0  16.0    16.0   

    maeduc  speduc          degree    ...           agecat   childcat news1  \
0     12.0     NaN     High school    ...         55 to 64     01-Feb    No   
1      NaN     NaN  Junior college    ...         25 to 34       None    No   
2     12.0    16.0     High school    ...         35 to 44     01-Feb    No   
3     12.0     NaN     High school    ...     Less than 25       None    No   
4     12.0     NaN        Bachelor    ...         35 to 44       None    No   
5      6.0     NaN        Bachelor    ...         25 to 34     01-Feb    No   
6     12.0     NaN     High school    ...         35 to 44       None    No   
7     12.0     NaN     High school    ...         25 to 34       None    No   
8     12.0    13.0        Bachelor    ...         35 to 44     01-Feb   Yes   
9     12.0     NaN  Junior college    ...         45 to 54       None    No   
10    12.0     NaN     High school    ...         25 to 34       None    No   
11     8.0    14.0     High school    ...         35 to 44     01-Feb    No   
12    12.0     NaN     High school    ...         25 to 34     01-Feb    No   
13     8.0    16.0        Bachelor    ...         45 to 54     01-Feb    No   
14    12.0     NaN     High school    ...     Less than 25       None    No   
15     6.0    12.0        Bachelor    ...         35 to 44  5 or more    No   
16     4.0     NaN  LT High school    ...         25 to 34     03-Apr    No   
17     NaN     NaN  LT High school    ...         25 to 34  5 or more    No   
18     NaN     NaN  LT High school    ...         35 to 44       None    No   
19    16.0    16.0        Bachelor    ...         25 to 34     01-Feb    No   

   news2  news3 news4 news5      car1      car2      car3  
0     No     No    No    No  American  Japanese  Japanese  
1     No    Yes    No    No  American    German  Japanese  
2     No     No   Yes   Yes  American  American       NaN  
3     No     No   Yes   Yes  American     Other       NaN  
4     No     No    No    No  American  American    Korean  
5    Yes     No   Yes   Yes  American    Korean  Japanese  
6     No     No   Yes   Yes    Korean  American  American  
7     No     No   Yes   Yes     Other  Japanese       NaN  
8     No    Yes    No    No  American    Korean       NaN  
9    Yes     No   Yes   Yes    Korean  Japanese    Korean  
10   Yes    Yes    No   Yes    Korean  Japanese  Japanese  
11    No     No    No    No  Japanese    German 

In [88]:
dataset.isnull().sum()

id             0
wrkstat        0
marital        0
childs         7
age            4
educ          12
paeduc       791
maeduc       433
speduc      1521
degree        10
sex            0
race           0
born          13
parborn       20
granborn     202
income       329
rincome      983
polviews     141
cappun       233
postlife     766
happy         26
hapmar      1495
owngun       963
news         962
tvhours      495
howpaid     1485
ethnic       594
eth1         275
eth2        1730
eth3        2388
confinan     968
conbus      1011
coneduc      951
conpress     970
conmedic     957
contv        956
agecat         4
childcat       7
news1          0
news2          0
news3          0
news4          0
news5          0
car1           0
car2         566
car3        1160
dtype: int64

a. Is there any relationship in between labour force status with marital status? 

###### Hypothesis testing  for relation between labour force (categorical data as per data set) and marital status(categorical data as per data set):

HO : labour force and marital status are independent
    
H1 : labour force and marital status are dependent
    
Since both the variables are categorical here , we will do a chi square test to test the relationship

In [93]:
t=pd.crosstab(dataset.wrkstat,dataset.marital,margins=True)
t

marital                  Divorced  Married  Never married  Separated  Widowed  \
wrkstat                                                                         
Keeping house                  25      200             35         13       55   
Other                          12       16             14          4        8   
Retired                        53      168             17          6      150   
School                          7        9             60          2        1   
Temporarily not working         9       23             11          1        2   
Unemployed, laid off           10       13             32          0        3   
Working full time             295      780            392         58       44   
Working part-time              35      138            102          9       20   
All                           446     1347            663         93      283   

marital                   All  
wrkstat                        
Keeping house             328  
Other                      54  
Retired                   394  
School                     79  
Temporarily not working    46  
Unemployed, laid off       58  
Working full time        1569  
Working part-time         304  
All                      2832

In [94]:
stats.chi2_contingency(observed=t)

(729.7434446415614,
 1.435325149487234e-127,
 40,
 array([[5.16553672e+01, 1.56008475e+02, 7.67881356e+01, 1.07711864e+01,
         3.27768362e+01, 3.28000000e+02],
        [8.50423729e+00, 2.56843220e+01, 1.26419492e+01, 1.77330508e+00,
         5.39618644e+00, 5.40000000e+01],
        [6.20494350e+01, 1.87400424e+02, 9.22394068e+01, 1.29385593e+01,
         3.93721751e+01, 3.94000000e+02],
        [1.24413842e+01, 3.75752119e+01, 1.84947034e+01, 2.59427966e+00,
         7.89442090e+00, 7.90000000e+01],
        [7.24435028e+00, 2.18792373e+01, 1.07690678e+01, 1.51059322e+00,
         4.59675141e+00, 4.60000000e+01],
        [9.13418079e+00, 2.75868644e+01, 1.35783898e+01, 1.90466102e+00,
         5.79590395e+00, 5.80000000e+01],
        [2.47095339e+02, 7.46272246e+02, 3.67318856e+02, 5.15243644e+01,
         1.56789195e+02, 1.56900000e+03],
        [4.78757062e+01, 1.44593220e+02, 7.11694915e+01, 9.98305085e+00,
         3.03785311e+01, 3.04000000e+02],
        [4.46000000e+02, 1.347

Since p value is <0.05 , thus we reject null

Thus we conclude that labour force and marital status have a replationship among them under confidence level of 95% and type 1 error(alpha)=0.05

a. Do you think educational qualification is somehow controlling the marital status? 

###### Hypothesis testing  for relation between educational qualification (categorical data as per data set) and marital status(categorical data as per data set):

HO : educational qualification and marital status are independent
    
H1 : educational qualification and marital status are dependent
    
Since both the variables are categorical here , we will do a chi square test to test the relationship

In [98]:
dataset.head(10)

id            wrkstat        marital  childs   age  educ  paeduc  maeduc  \
0   1  Working full time       Divorced     2.0  60.0  12.0    12.0    12.0   
1   2  Working part-time  Never married     0.0  27.0  17.0    20.0     NaN   
2   3  Working full time        Married     2.0  36.0  12.0    12.0    12.0   
3   4  Working full time  Never married     0.0  21.0  13.0     NaN    12.0   
4   5  Working full time  Never married     0.0  35.0  16.0     NaN    12.0   
5   6  Working full time       Divorced     1.0  33.0  16.0     9.0     6.0   
6   7  Working full time      Separated     0.0  43.0  12.0    14.0    12.0   
7   8  Working full time  Never married     0.0  29.0  13.0    16.0    12.0   
8   9  Working part-time        Married     2.0  39.0  18.0    16.0    12.0   
9  10  Working full time       Divorced     0.0  45.0  15.0    16.0    12.0   

   speduc          degree    ...           agecat childcat news1 news2  news3  \
0     NaN     High school    ...         55 to 64   01-Feb    No    No     No   
1     NaN  Junior college    ...         25 to 34     None    No    No    Yes   
2    16.0     High school    ...         35 to 44   01-Feb    No    No     No   
3     NaN     High school    ...     Less than 25     None    No    No     No   
4     NaN        Bachelor    ...         35 to 44     None    No    No     No   
5     NaN        Bachelor    ...         25 to 34   01-Feb    No   Yes     No   
6     NaN     High school    ...         35 to 44     None    No    No     No   
7     NaN     High school    ...         25 to 34     None    No    No     No   
8    13.0        Bachelor    ...         35 to 44   01-Feb   Yes    No    Yes   
9     NaN  Junior college    ...         45 to 54     None    No   Yes     No   

  news4 news5      car1      car2      car3  
0    No    No  American  Japanese  Japanese  
1    No    No  American    German  Japanese  
2   Yes   Yes  American  American       NaN  
3   Yes   Yes  American     Other       NaN  
4    No    No  American  American    Korean  
5   Yes   Yes  American    Korean  Japanese  
6   Yes   Yes    Korean  American  American  
7   Yes   Yes     Other  Japanese       NaN  
8    No    No  American    Korean       NaN  
9   Yes   Yes    Korean  Japanese    Korean  

[10 rows x 46 columns]

In [99]:
dataset.degree.mode()[0]

'High school'

In [100]:
dataset.degree.fillna(dataset.degree.mode()[0],inplace=True)

In [102]:
dataset.head(10)

id            wrkstat        marital  childs   age  educ  paeduc  maeduc  \
0   1  Working full time       Divorced     2.0  60.0  12.0    12.0    12.0   
1   2  Working part-time  Never married     0.0  27.0  17.0    20.0     NaN   
2   3  Working full time        Married     2.0  36.0  12.0    12.0    12.0   
3   4  Working full time  Never married     0.0  21.0  13.0     NaN    12.0   
4   5  Working full time  Never married     0.0  35.0  16.0     NaN    12.0   
5   6  Working full time       Divorced     1.0  33.0  16.0     9.0     6.0   
6   7  Working full time      Separated     0.0  43.0  12.0    14.0    12.0   
7   8  Working full time  Never married     0.0  29.0  13.0    16.0    12.0   
8   9  Working part-time        Married     2.0  39.0  18.0    16.0    12.0   
9  10  Working full time       Divorced     0.0  45.0  15.0    16.0    12.0   

   speduc          degree    ...           agecat childcat news1 news2  news3  \
0     NaN     High school    ...         55 to 64   01-Feb    No    No     No   
1     NaN  Junior college    ...         25 to 34     None    No    No    Yes   
2    16.0     High school    ...         35 to 44   01-Feb    No    No     No   
3     NaN     High school    ...     Less than 25     None    No    No     No   
4     NaN        Bachelor    ...         35 to 44     None    No    No     No   
5     NaN        Bachelor    ...         25 to 34   01-Feb    No   Yes     No   
6     NaN     High school    ...         35 to 44     None    No    No     No   
7     NaN     High school    ...         25 to 34     None    No    No     No   
8    13.0        Bachelor    ...         35 to 44   01-Feb   Yes    No    Yes   
9     NaN  Junior college    ...         45 to 54     None    No   Yes     No   

  news4 news5      car1      car2      car3  
0    No    No  American  Japanese  Japanese  
1    No    No  American    German  Japanese  
2   Yes   Yes  American  American       NaN  
3   Yes   Yes  American     Other       NaN  
4    No    No  American  American    Korean  
5   Yes   Yes  American    Korean  Japanese  
6   Yes   Yes    Korean  American  American  
7   Yes   Yes     Other  Japanese       NaN  
8    No    No  American    Korean       NaN  
9   Yes   Yes    Korean  Japanese    Korean  

[10 rows x 46 columns]

In [103]:
dataset.isnull().sum()

id             0
wrkstat        0
marital        0
childs         7
age            4
educ          12
paeduc       791
maeduc       433
speduc      1521
degree         0
sex            0
race           0
born          13
parborn       20
granborn     202
income       329
rincome      983
polviews     141
cappun       233
postlife     766
happy         26
hapmar      1495
owngun       963
news         962
tvhours      495
howpaid     1485
ethnic       594
eth1         275
eth2        1730
eth3        2388
confinan     968
conbus      1011
coneduc      951
conpress     970
conmedic     957
contv        956
agecat         4
childcat       7
news1          0
news2          0
news3          0
news4          0
news5          0
car1           0
car2         566
car3        1160
dtype: int64

In [104]:
t=pd.crosstab(dataset.degree,dataset.marital,margins=True)
t

marital         Divorced  Married  Never married  Separated  Widowed   All
degree                                                                    
Bachelor              58      251            129         12       28   478
Graduate              29      123             41          3        9   205
High school          244      690            370         58      148  1510
Junior college        45      109             46          3        6   209
LT High school        70      174             77         17       92   430
All                  446     1347            663         93      283  2832

In [105]:
stats.chi2_contingency(observed=t)

(123.18796051189827,
 6.0450187703046e-15,
 25,
 array([[  75.27824859,  227.35381356,  111.90466102,   15.6970339 ,
           47.76624294,  478.        ],
        [  32.28460452,   97.50529661,   47.99258475,    6.73199153,
           20.4855226 ,  205.        ],
        [ 237.80367232,  718.20974576,  353.50635593,   49.58686441,
          150.89336158, 1510.        ],
        [  32.91454802,   99.40783898,   48.92902542,    6.86334746,
           20.88524011,  209.        ],
        [  67.71892655,  204.52330508,  100.66737288,   14.12076271,
           42.96963277,  430.        ],
        [ 446.        , 1347.        ,  663.        ,   93.        ,
          283.        , 2832.        ]]))

Since p value <0.05, we reject the null 

Thus educational qualification is controlling the marital status with confidence level of 95% and type 1 error(alpha)=0.05

c. Is happiness is driven by earnings or marital status? 

case 1:
    
###### Hypothesis testing  for relation between happiness (categorical data as per data set) and marital status(categorical data as per data set):

HO : happiness and marital status are independent
    
H1 : happiness and marital status are dependent
    
Since both the variables are categorical here , we will do a chi square test to test the relationship


case 2:
    
###### Hypothesis testing  for relation between happiness (categorical data as per data set) and earnings(categorical data as per data set):

HO : happiness and earnings are independent
    
H1 : happiness and earnings are dependent
    
Since both the variables are categorical here , we will do a chi square test to test the relationship

In [107]:
dataset.income.mode()[0]

'$25000 or more'

In [108]:
dataset.happy.mode()[0]

'Pretty happy'

In [109]:
dataset.income.head(10)

0    $25000 or more
1    $15000 - 19999
2    $25000 or more
3    $15000 - 19999
4    $25000 or more
5    $20000 - 24999
6    $25000 or more
7    $25000 or more
8               NaN
9    $25000 or more
Name: income, dtype: object

In [110]:
dataset.happy.head(10)

0    Pretty happy
1    Pretty happy
2      Very happy
3      Very happy
4    Pretty happy
5    Pretty happy
6      Very happy
7    Pretty happy
8      Very happy
9    Pretty happy
Name: happy, dtype: object

In [111]:
dataset.income.fillna(dataset.income.mode()[0],inplace=True)
dataset.happy.fillna(dataset.happy.mode()[0],inplace=True)

In [113]:
dataset.income.head(10)

0    $25000 or more
1    $15000 - 19999
2    $25000 or more
3    $15000 - 19999
4    $25000 or more
5    $20000 - 24999
6    $25000 or more
7    $25000 or more
8    $25000 or more
9    $25000 or more
Name: income, dtype: object

In [112]:
dataset.happy.head(10)

0    Pretty happy
1    Pretty happy
2      Very happy
3      Very happy
4    Pretty happy
5    Pretty happy
6      Very happy
7    Pretty happy
8      Very happy
9    Pretty happy
Name: happy, dtype: object

In [114]:
dataset.isnull().sum()

id             0
wrkstat        0
marital        0
childs         7
age            4
educ          12
paeduc       791
maeduc       433
speduc      1521
degree         0
sex            0
race           0
born          13
parborn       20
granborn     202
income         0
rincome      983
polviews     141
cappun       233
postlife     766
happy          0
hapmar      1495
owngun       963
news         962
tvhours      495
howpaid     1485
ethnic       594
eth1         275
eth2        1730
eth3        2388
confinan     968
conbus      1011
coneduc      951
conpress     970
conmedic     957
contv        956
agecat         4
childcat       7
news1          0
news2          0
news3          0
news4          0
news5          0
car1           0
car2         566
car3        1160
dtype: int64

In [116]:
## case 1:
    
t=pd.crosstab(dataset.happy,dataset.marital,margins=True) 
t

marital        Divorced  Married  Never married  Separated  Widowed   All
happy                                                                    
Not too happy        72       71            108         30       59   340
Pretty happy        281      694            435         50      141  1601
Very happy           93      582            120         13       83   891
All                 446     1347            663         93      283  2832

In [117]:
stats.chi2_contingency(observed=t)

(261.0091307378115,
 6.6685158886813e-47,
 15,
 array([[  53.54519774,  161.71610169,   79.59745763,   11.16525424,
           33.9759887 ,  340.        ],
        [ 252.13488701,  761.49258475,  374.81038136,   52.57521186,
          159.98693503, 1601.        ],
        [ 140.31991525,  423.79131356,  208.59216102,   29.2595339 ,
           89.03707627,  891.        ],
        [ 446.        , 1347.        ,  663.        ,   93.        ,
          283.        , 2832.        ]]))

Since p value<0.0.5 we reject the null, hence happiness is dependent on marital status

In [118]:
## case 2:
    
t=pd.crosstab(dataset.happy,dataset.income,margins=True) 
t

income         $1000 TO 2999  $10000 - 14999  $15000 - 19999  $20000 - 24999  \
happy                                                                          
Not too happy              7              39              33              40   
Pretty happy              20             109             120             157   
Very happy                 5              44              26              50   
All                       32             192             179             247   

income         $25000 or more  $3000 TO 3999  $4000 TO 4999  $5000 TO 5999  \
happy                                                                        
Not too happy             151              9              9              6   
Pretty happy             1074             11             13             18   
Very happy                691              4             10             11   
All                      1916             24             32             35   

income         $6000 TO 6999  $7000 TO 7999  $8000 TO 9999  LT $1000   All  
happy                                                                       
Not too happy             14             12              9        11   340  
Pretty happy              13             21             31        14  1601  
Very happy                 6             14             19        11   891  
All                       33             47             59        36  2832

In [119]:
stats.chi2_contingency(observed=t)

(177.41659890776808,
 1.3483634070060814e-20,
 36,
 array([[   3.84180791,   23.05084746,   21.49011299,   29.6539548 ,
          230.02824859,    2.88135593,    3.84180791,    4.2019774 ,
            3.96186441,    5.64265537,    7.08333333,    4.3220339 ,
          340.        ],
        [  18.09039548,  108.54237288,  101.19314972,  139.63524011,
         1083.16242938,   13.56779661,   18.09039548,   19.78637006,
           18.65572034,   26.57026836,   33.35416667,   20.35169492,
         1601.        ],
        [  10.06779661,   60.40677966,   56.31673729,   77.71080508,
          602.80932203,    7.55084746,   10.06779661,   11.01165254,
           10.38241525,   14.78707627,   18.5625    ,   11.32627119,
          891.        ],
        [  32.        ,  192.        ,  179.        ,  247.        ,
         1916.        ,   24.        ,   32.        ,   35.        ,
           33.        ,   47.        ,   59.        ,   36.        ,
         2832.        ]]))

Since p value<0.0.5 we reject the null, hence happiness is dependent on income

From the hypothesis testing we can see that happiness is both dependent on marital status and income


But 

statistics value of happiness and marital status =261.0091307378115

                    is greater than 

statistics value of happiness and income =177.41659890776808


Thus we can say happiness is more dependent on marital status with confidence level of 95% and type 1 error (aplha)=0.05